src/tfr_executive/src/autonomous_action_server.cpp

```diff
@@ -80,39 +124,189 @@ class AutonomousExecutive
+                        server.setPreempted();
+                        ROS_INFO("Autonomous Action Server: navigation preempted");
+                        return;
+                    }
+                    frequency.sleep();
+                }
 
-            ROS_INFO("Autonomous Action Server: commencing navigation");
-            ROS_INFO("Autonomous Action Server: navigation finished");
+                if (navigationClient.getState()!=actionlib::SimpleClientGoalState::SUCCEEDED)
+                {
+                    ROS_INFO("Autonomous Action Server: navigation to failed");
+                    server.setAborted();
+                    return;
+                }
+                ROS_INFO("Autonomous Action Server: navigation finished");
+            }
+            if (DUMPING)
+            {
+                ROS_INFO("Autonomous Action Server: Connecting to dumping server");
+                navigationClient.waitForServer();
+                ROS_INFO("Autonomous Action Server: Connected to dumping server");
 
-            ROS_INFO("Autonomous Action Server: commencing dumping");
-            ROS_INFO("Autonomous Action Server: dumping finished");
+                ROS_INFO("Autonomous Action Server: commencing dumping");
+ 
+                tfr_msgs::EmptyGoal goal;
+                dumpingClient.sendGoal(goal);
+                //handle preemption
+                while ( !dumpingClient.getState().isDone() && ros::ok())
+                {
+                    if (server.isPreemptRequested() || ! ros::ok())
+                    {
+                        dumpingClient.cancelAllGoals();
+                        server.setPreempted();
+                        ROS_INFO("Autonomous Action Server: dumping preempted");
+                        return;
+                    }
+                    frequency.sleep();
+                }
 
-            tfr_msgs::EmptyResult result{};
-            server.setSucceeded(result);
-            ROS_INFO("Autonomous Action Server: mission finished");
+                if (dumpingClient.getState()!=actionlib::SimpleClientGoalState::SUCCEEDED)
+                {
+                    ROS_INFO("Autonomous Action Server: dumping failed");
+                    server.setAborted();
+                    return;
+                }
+                ROS_INFO("Autonomous Action Server: dumping finished");
+
+            }
+            ROS_INFO("Autonomous Action Server: AUTONOMOUS MISSION SUCCESS");
+            server.setSucceeded();
         }
 
         actionlib::SimpleActionServer<tfr_msgs::EmptyAction> server;
+
+        actionlib::SimpleActionClient<tfr_msgs::EmptyAction> localizationClient;
+        actionlib::SimpleActionClient<tfr_msgs::NavigationAction> navigationClient;
+        actionlib::SimpleActionClient<tfr_msgs::DiggingAction> diggingClient;
+        actionlib::SimpleActionClient<tfr_msgs::EmptyAction> dumpingClient;
+
+        bool LOCALIZATION;
```

Comment:
Do we want to make these uppercase like this? It doesn't seem consistent with the TFR coding style.